docs/design/2020-05-08-standardize-error-codes-and-messages.md

```diff
@@ -162,35 +162,66 @@ In addition, the error codes should be append only in case of conflict between v
 ### The Error Definition
 
 In the discussion above, an error has at least 4 parts:
-- The error code: it's the identity of an error
-- The error field: it's the error itself the user can view in TiDB system
-- the message field: the description of the error, what happened and why happend?
-- the workaround filed: how to workaround this error
+- The error code: it's the identity of an error.
+- The error field: it's the error itself the user can view in TiDB system. Like `err.Error()`.
+- The description field: the expanded detail of why this error occurred. This could be written by developer outside the code, and the more detail this field explaining the better, even some guess of the cause could be included.
+- The workaround filed: how to work around this error. It's used to teach the users how to solve the error if occurring in the real environment.
 
 Besides, we can append a optional tags field to it:
 ```toml
 [error.9005]
 error = ""
-message = ""
+description = ""
 workaround = ""
 tags = ["tikv"]
 ```
 
-The tags is used to classify errors (e.g. the level of seriousness). At the very beginning, we can ignore it since we don't have enough errors listed. Once we have enough data, we need to classify all errors by different dimensions. Then we will make out a standard abount how to classify errors.
+The tags is used to classify errors (e.g. the level of seriousness). At the very beginning, we can ignore it since we don't have enough errors listed. Once we have enough data, we need to classify all errors by different dimensions. Then we will make out a standard about how to classify errors.
 
 #### The Error Code Range
 
+The error code is a 3-tuple of abbreviated component name, error class and error code, joined by a colon like `{Component}:{ErrorClass}:{ErrorCode}`.
+
+Where `Component` field is the abbreviated component name of the error source, wrote as upper case, component names are mapped as below:
+
+- TiKV: KV
+- PD: PD
+- DM: DM
+- BR: BR
+- TiCDC: CDC
+- Lightning: LN
+- TiFlash: FLASH
+- Dumpling: DP
+
+The `ErrorClass` is the name of the `ErrClass` the error belongs to, which defined by `[errors.RegisterErrorClass](https://github.com/pingcap/errors/blob/f9054262e67a3704a936a6ea216e73287486490d/terror/terror.go#L41)` or someway likewise. If this is unacceptable (for projects not written with golang), anything that can classify the "type" of this error (e.g., package name.) would also be good. 
+
+The `ErrorCode` is the identity of this error. Both numeric and textual code are acceptable, but it would be better to provide textual code, which should be one or two short words with PascalCase named to identity the error.
```

Comment:
Can we provide a detailed example of the whole log?

Reply:
What's that? Do you mean an example of generating rule of the code...?

Reply:
just an example of the generated log.

Reply:
"the log" means error.toml? or [ERROR] balabala....?

Reply:
`[2020/07/21 19:20:23.494 +08:00] [ERROR] [xxx.go:100] ["xxxxxxx"]`

Reply:
Done.